src/zonetree/answer.rs

```diff
@@ -0,0 +1,216 @@
+//! Answers to zone tree queries.
+
+//------------ Answer --------------------------------------------------------
+
+use super::{SharedRr, SharedRrset, StoredDname};
+use crate::base::iana::Rcode;
+use crate::base::message_builder::AdditionalBuilder;
+use crate::base::wire::Composer;
+use crate::base::Message;
+use crate::base::MessageBuilder;
+use octseq::Octets;
+
+/// A DNS answer to a query against a [`Zone`].
+///
+/// [`Answer`] is the type returned by [`Zone::query()`].
+///
+/// Callers of [`Zone::query()`] will likely only ever need to use the
+/// [`Self::to_message()`] function. Alternatively, for complete control use
+/// the getter functions on [`Answer`] instead and construct a response
+/// message yourself using [`MessageBuilder`].
+///
+/// Implementers of alternate backing stores for [`Zone`]s will need to use
+/// one of the various `Answer` constructor functions when [`Zone::query()`]
```

Comment:
Apparently, the preferred way to refer to functions is without the parentheses.

Reply:
I will change it, though fns without parens look to me. For my education, what are you basing that on? Is it because of this in the rustdoc book? From: https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html

> You can also disambiguate for functions by adding () after the function name, or for macros by adding ! after the macro name. The macro ! can be followed by (), {}, or []. Example:

Reply:
FYI other existing code in domain already uses the`[xxx()]` rustdoc syntax, e.g. `OptRcode`.

Reply:
Done, in all places I found it both in code changed in this branch and elsewhere to make it consistent.

Reply:
I think this is (or was at some point) in one of the style guides. Can’t find it right now. But this is the style that e.g. the std library uses.